Yoast/wordpress-seo
# Add meta description date

## Summary

This PR can be summarized in the following changelog entry:

* N/A

## Relevant technical choices:

* Added the optional meta description date to the React SnippetPreview.
* Depends on PR: https://github.com/Yoast/yoast-components/pull/484

## Test instructions

This PR can be tested by following these steps:

* Install deps: `composer install & yarn install`.
* Link the yoast-components branch:
  * In the yoast-components clone: `yarn link` (skip this step if already done before).
  * In wordpress-seo clone: `yarn link yoast-components`.
* Switch yoast-components to the required branch: `stories/add-date-to-snippet-editor-props`.
* Install and build yoast-components: `yarn install && yarn start`.
* Start the wordpress-seo dev server: `yarn start`.
* In the `wp-config.php` file of your WordPress installation add `define( 'YOAST_FEATURE_SNIPPET_PREVIEW', true );` to enable the React SnippetPreview.
* On the `SEO -> Search Appearance -> Content Types` page enable the option: `Date in Snippet Preview`.
* On the edit of posts & pages there should now be date in front of the meta description.
* Check if the option enable/disables the date in the React SnippetPreview just like the old version.

## Quality assurance

~~* [ ] I have added unittests to verify the code works as intended~~

Fixes #9535
